enrichment_method.py

```diff
@@ -126,26 +128,31 @@ def lambda_handler(event, context):
     return final_output
 
 
-def marine_mismatch_detector(data, check_column, period_column, identifier_column):
+def marine_mismatch_detector(data,
+                             survey_column,
+                             check_column,
+                             period_column,
+                             identifier_column):
     """
     Detects references that are producing marine but from a county that doesnt produce marine  # noqa: E501
     :param data: Input data after having been merged with responder_county_lookup - DataFrame
+    :param survey_column: Survey code value - String
     :param check_column: column to check against(marine) - String
     :param period_column: Column that holds the period - String
     :param identifier_column: Column that holds the unique id of a row(usually responder id) - String
     :return: bad_data_with_marine: Df containing information about any reference that is producing marine when it shouldn't - DataFrame
     """
 
     bad_data = data[
-        (data["land_or_marine"] == "M")
+        (survey_column == "076")
         & (data[check_column] == "n")
         ]
     bad_data["issue"] = "Reference should not produce marine data"
     return bad_data[
         [
             identifier_column,
             "issue",
-            "land_or_marine",
+            "survey",

```

Comment:
This appears to be a hardcoded column name. Should this instead be the survey_column variable?

Reply:
No as then it would just be the value and it was hardcoded to land or marine last time. So was just changing to match but unsure of what's best.

Reply:
Isn't what we pass in the name of the survey code column, so the code can look at the column and check if it is the land survey code or marine survey code?

Reply:
Ah, I see what you mean. Let me have a fiddle.

Reply:
Resolved.